lib/src/Control/Arrow/Error.hs

```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE Arrows #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE UndecidableInstances #-}
+module Control.Arrow.Error where
+
+import Control.Arrow
+
+import Control.Monad.TryWithError
+
+class Arrow c => ArrowError e x y c where
+  tryWithErrorA :: c x y -> c y y -> c e y -> c x y
+
+instance MonadTryWithError m => ArrowError String x y (Kleisli m) where
```

Comment:
Please do not repeat the arrow type classes as monad type classes. We have enough boilerplate code already.